lib/rules/the-step-down-rule.js

```diff
@@ -7,9 +8,36 @@ module.exports = {
   create(context) {
     return {
       Program() {
+        methodStack = [];
         requiredFunctions = []; // Clearing list between runs
         findVariablesInScope(context.getScope(), context);
       },
+      MethodDefinition(node) {
+        if (node.kind === 'method') {
+          methodStack.push(node.key.name);
+        }
+      },
+      'CallExpression:exit'(node) {
+        if (node.callee.type !== 'MemberExpression' || node.callee.property.type !== 'Identifier') {
+          return;
+        }
+
+        const isThisCall = node.callee.object.type === 'ThisExpression';
+        if (!isThisCall) {
+          return;
+        }
+
+        const calledMethodName = node.callee.property.name;
+        const callerMethodNameIndex = methodStack.indexOf(calledMethodName);
+        if (callerMethodNameIndex === -1 || callerMethodNameIndex >= methodStack.length - 1) {
```

Comment:
Nit: Would have liked to see a named variable here like you did with `isThisCall`, but in no way a showstopper.

Reply:
I agree, that would have been nicer